src/image.rs

```diff
@@ -377,12 +377,35 @@ pub(crate) fn decoder_to_vec<'a>(decoder: impl ImageDecoder<'a>) -> ImageResult<
 }
 
 /// Represents the progress of an image operation.
+///
+/// Note that this is not necessarily accurate and no change to the values passed to the progress
+/// function during decoding will be considered breaking. A decoder could in theory report the
+/// progress `(0, 0)` if progress is unknown, without violating the interface contract of the type.
```

Comment:
I don't see the point of saying this. If a decoder doesn't support reporting progress, it can just do a callback with (0, total_bytes) before it starts and (total_bytes, total_bytes) after it finishes.

Reply:
What you describe *is* a form or progress reporting and it requires the decoder to know `total_bytes`. For a compressed stream, which are the total bytes? The bytes to represent the image or the bytes of the compressed stream?

I just want to say that you should be extra cautious and leave us more freedom on whether we implement this for a decoder. Note that if we were to not leave this open, we need to asnwer the above question to provide progress reports for decoders that do not have such a callback in their own implementation.